src/rpc/rawtransaction.cpp

```diff
@@ -1775,6 +1814,118 @@ UniValue rawissueasset(const JSONRPCRequest& request)
+
+    CMutableTransaction mtx;
+
+    if (!DecodeHexTx(mtx, request.params[0].get_str()))
+        throw JSONRPCError(RPC_DESERIALIZATION_ERROR, "TX decode failed");
+
+    UniValue issuances = request.params[1].get_array();
+
+    unsigned int num_issuances = 0;
+
+    for (unsigned int idx = 0; idx < issuances.size(); idx++) {
+        const UniValue& issuance = issuances[idx];
+        const UniValue& issuance_o = issuance.get_obj();
+
+        CAmount asset_amount = 0;
+        const UniValue& asset_amount_uni = issuance_o["asset_amount"];
+        if (asset_amount_uni.isNum()) {
+            asset_amount = AmountFromValue(asset_amount_uni);
+            if (asset_amount <= 0) {
+                throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid parameter, asset_amount must be positive");
+            }
+        } else {
+            throw JSONRPCError(RPC_INVALID_PARAMETER, "Asset amount must be given for each reissuance.");
+        }
+
+        const UniValue& asset_address_uni = issuance_o["asset_address"];
+        if (!asset_address_uni.isStr()) {
+            throw JSONRPCError(RPC_INVALID_PARAMETER, "Reissuance missing asset_address");
+        }
+        std::string asset_address_str = asset_address_uni.get_str();
+
+        int input_index = -1;
+        const UniValue& input_index_o = issuance_o["input_index"];
+        if (input_index_o.isNum()) {
+            input_index = input_index_o.get_int();
+            if (input_index < 0) {
+                throw JSONRPCError(RPC_INVALID_PARAMETER, "Input index must be non-negative.");
+            }
+        } else {
+            throw JSONRPCError(RPC_INVALID_PARAMETER, "Input indexes for all reissuances are required.");
+        }
+
+        if (!issuance_o["asset_blinder"].isStr()) {
+            throw JSONRPCError(RPC_INVALID_PARAMETER, "Asset blinders for all reissuances are required.");
+        }
+        std::string asset_blinder_str = issuance_o["asset_blinder"].get_str();
+        if (!IsHex(asset_blinder_str)) {
+            throw JSONRPCError(RPC_INVALID_PARAMETER, "Asset blinders must be hex strings.");
+        }
+
+        uint256 asset_blinder;
+        asset_blinder.SetHex(asset_blinder_str);
+
+        std::string entropy_str = issuance_o["entropy"].get_str();
+        if (!IsHex(entropy_str) || entropy_str.size() != 64) {
+            throw JSONRPCError(RPC_INVALID_PARAMETER, "Entropy must be 64 character hex strings.");
+        }
+
+        uint256 entropy;
+        entropy.SetHex(entropy_str);
+
+        // Optional, default of 0's
+        uint256 contract_hash;
+        if (issuance_o["contract_hash"].isStr()) {
+            std::string contract_hash_str = issuance_o["contract_hash"].get_str();
+            if (!IsHex(contract_hash_str) || contract_hash_str.size() != 64) {
+                throw JSONRPCError(RPC_INVALID_PARAMETER, "Contract hash must be 64 character hex strings.");
+            }
+            contract_hash = uint256(ParseHex(contract_hash_str));
```

Comment:
We should be using `ParseHashV` to make sure we're parsing it like a hash like everywhere else in Bitcoin Core and to do argument checking.